Assets/MixedRealityToolkit/Attributes/MixedRealityExtensionServiceAttribute.cs

```diff
@@ -33,6 +33,11 @@ public class MixedRealityExtensionServiceAttribute : Attribute
         /// </summary>
         public virtual SupportedPlatforms RuntimePlatforms { get; }
 
+        /// <summary>
+        /// Is a Profile explicitly required?
```

Comment:
Should we have documentation or comments on what happens if a profile isn't given but is required?

Reply:
That would vary based on system implementation. Some will generate null reference exceptions, others may not behave as expected.

I can add "The data provider may not behave as expected without the required profile." to the warning message if that feels right.

Reply:
Also do we care about enforcing this at runtime?

Reply:
I can also add additional doc comments here to try to educate service implementors on the need for reasonable behavior / proper marking of this value.

Reply:
> Also do we care about enforcing this at runtime?

We _could_ consider hiding the profile field if requires profile is set to false. That would be a separate change, however.

If i am misunderstanding the question, please let me know.

Reply:
As far as the specific behavior of a given provider, the behavior is up to the the developer. I'm not sure what we could do to enforce a behavior.

Reply:
Comments concern:
I am thinking more as a person who is reading the documentation and leveraging this attribute. A remark saying what I should do if a profile is null or not provided but I require it would be helpful. Guidance on how should I fail

Enforcement:
Enforcement at runtime could be done at registration whether by the data provider class itself (which we could maybe put in a base class) or by the service registering the data provider (like the Input System)

Reply:
Like SceneTransitionService would just fail it looks like with a null exception
```
/// <inheritdoc />
        public override void Initialize()
        {
            UseFadeColor = sceneTransitionServiceProfile.UseFadeColor;
            FadeColor = sceneTransitionServiceProfile.FadeColor;
            FadeInTime = sceneTransitionServiceProfile.FadeInTime;
            FadeOutTime = sceneTransitionServiceProfile.FadeOutTime;
            FadeTargets = sceneTransitionServiceProfile.FadeTargets;
        }
```

Thus we could add in like the base dataprovider/extension service class a check in the constructor if I require a profile but the one given was null

Reply:
Just a thought to consider

Reply:
> we could add in like the base dataprovider/extension service class a check in the constructor if I require a profile but the one given was null

Great thought! That way, the null refs would make sense to the customer. I like it!

Reply:
Added comment to the attribute docs

```
        /// <remarks>
        /// Services requiring a configuration profile should include code that helps inform customers
        /// if one is not provided. A check in the service constructor with a corresponding warning log
        /// message is recommended.
        /// </remarks>
```